src/utils/signer.ts

```diff
@@ -304,6 +306,33 @@ export async function signRequestV4(
   return newReq;
 }
 
+/**
+ * Verifies if an IP address is within the range of any of the given CIDR addresses.
+ * @param ipAddress The IP address to check (e.g., "192.168.1.10").
+ * @param cidrAddresses An array of CIDR address ranges (e.g., ["192.168.1.0/24", "10.0.0.0/8"]).
+ * @returns True if the IP address is within any of the CIDR ranges, false otherwise.
+ */
+export function isIpInMultipleCidrRanges(
+  ipAddress: string,
+  cidrAddresses: string[],
+): boolean {
+  for (const cidrAddress of cidrAddresses) {
+    // Validate the CIDR address first
+    if (!IPCIDR.isValidCIDR(cidrAddress)) {
```

Comment:
this should warn/error just as the configuration passes, otherwise you will have silenced error